### PR TITLE
add types chromium-installation-events

### PR DESCRIPTION
**Pull request by dartess**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tslint.json` [should contain](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#linter-tslintjson) `{ "extends": "@definitelytyped/dtslint/dt.json" }`, and no additional rules.
- [x] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

**Comment by FluorescentHallucinogen** [comment]
@mgiuca @dmurph Could you please take a look?

File: types/dom-chromium-installation-events/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+export type AppBannerPromptOutcome = 'accepted' | 'dismissed';
+
+export interface PromptResponseObject {
+    readonly outcome: AppBannerPromptOutcome;
+    readonly platform: string;
+}
+
+declare global {
+    class BeforeInstallPromptEvent extends Event {
+        constructor(type: string, eventInitDict?: EventInit);
```
Comment (peterblazejewicz) [inline types/dom-chromium-installation-events/index.d.ts:15]: this one is documented as parameterless:
https://developer.mozilla.org/en-US/docs/Web/API/BeforeInstallPromptEvent#constructor

Comment (peterblazejewicz) [reply types/dom-chromium-installation-events/index.d.ts:15]: assuming convention for custom events

File: types/dom-chromium-installation-events/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,30 @@
+// Type definitions for non-npm package dom-chromium-installation-events 101.0
+// Project: https://developer.mozilla.org/en-US/docs/Web/API/BeforeInstallPromptEvent
+// Definitions by: Sergey Kozlov <https://github.com/dartess>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+export type AppBannerPromptOutcome = 'accepted' | 'dismissed';
```
Comment (peterblazejewicz) [inline types/dom-chromium-installation-events/index.d.ts:6]: this is global script, so you can write this eg. that way, without exports:

```ts
type AppBannerPromptOutcome = 'accepted' | 'dismissed';

interface PromptResponseObject {
    readonly outcome: AppBannerPromptOutcome;
    readonly platform: string;
}

interface BeforeInstallPromptEvent extends Event {
    readonly platforms: string[];
    readonly userChoice: Promise<PromptResponseObject>;
    prompt(): Promise<void>;
}

declare var BeforeInstallPromptEvent: {
    prototype: BeforeInstallPromptEvent;
    new (type: string, eventInitDict: EventInit): BeforeInstallPromptEvent;
};

interface WindowEventMap {
    beforeinstallprompt: BeforeInstallPromptEvent;
    appinstalled: Event;
}

interface Window {
    onappinstalled?: ((this: Window, ev: Event) => any) | null;
    onbeforeinstallprompt?: ((this: Window, ev: BeforeInstallPromptEvent) => any) | null;
}
```

Comment (dartess) [reply types/dom-chromium-installation-events/index.d.ts]: thanks, applied

**Comment by dartess** [comment]
please let me know if you can be added as an owner.

**Review by tomayac — APPROVED** [review]
LGTM

Thanks for adding these types. If you wanted to, you could add the possible platforms as an enum (rather than a `string[]`), they're listed in the [manifest spec](https://github.com/w3c/manifest/wiki/Platforms).

**Comment by dartess** [comment]
@tomayac I did it inline so as not to occupy any names globally.

**Comment by dartess** [comment]
Couldn't use inline due to tests.
I'll try to separate into a type a little later.

File: types/dom-chromium-installation-events/index.d.ts [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+type AppBannerPromptOutcome = 'accepted' | 'dismissed';
+
+interface PromptResponseObject {
+    readonly outcome: AppBannerPromptOutcome;
+    readonly platform: string;
+}
+
+type BeforeInstallPromptEventPlatform =
```
Comment (tomayac) [inline types/dom-chromium-installation-events/index.d.ts:13]:
```suggestion
// https://github.com/w3c/manifest/wiki/Platforms
type BeforeInstallPromptEventPlatform =
```

**Review by tomayac — APPROVED** [review]
LGTM

Just a nit to add the source for the platforms.

**Comment by amcasey** [comment]
LGTM.  @peterblazejewicz ?

**Review by peterblazejewicz — APPROVED** [review]
LGTM!

**Review by tomayac — APPROVED** [review]
LGTM

**Comment by dartess** [comment]
Ready to merge

**Pull request merged**